spec/acts_as_tag_pgarray/taggable_spec.rb

```diff
@@ -1,162 +1,188 @@
+      end
 
-  describe "#colors_cloud" do
-    it "returns tag cloud for tag_name" do
-      expect(User.colors_cloud).to match_array(
-        [@user1, @user2, @user3, @admin1, @admin2].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }
-      )
-    end
+      it "returns filtered tag cloud for tag_name with block" do
+        expect(User.colors_cloud { where(name: ["Ken", "Joe"]) }).to match_array(
+          [@user2, @user3].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }
+        )
+      end
 
-    it "returns filtered tag cloud for tag_name with block" do
-      expect(User.colors_cloud { where(name: ["Ken", "Joe"]) }).to match_array(
-        [@user2, @user3].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }
-      )
-    end
+      it "returns filtered tag cloud for tag_name with prepended scope" do
+        expect(User.where("tag like ?", "bl%").colors_cloud).to match_array(
+          [@user1, @user2, @user3, @admin2].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }.select { |name, count| name.start_with? "bl" }
+        )
+      end
 
-    it "returns filtered tag cloud for tag_name with prepended scope" do
-      expect(User.where("tag like ?", "bl%").colors_cloud).to match_array(
-        [@user1, @user2, @user3, @admin2].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }.select { |name, count| name.start_with? "bl" }
-      )
+      it "returns filtered tag cloud for tag_name with prepended scope and block" do
+        expect(User.where("tag like ?", "bl%").colors_cloud { where(name: ["Ken", "Joe"]) }).to match_array(
+          [@user2, @user3].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }.select { |name, count| name.start_with? "bl" }
+        )
+      end
     end
 
-    it "returns filtered tag cloud for tag_name with prepended scope and block" do
-      expect(User.where("tag like ?", "bl%").colors_cloud { where(name: ["Ken", "Joe"]) }).to match_array(
-        [@user2, @user3].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }.select { |name, count| name.start_with? "bl" }
-      )
+    describe "with complex scope" do
+      it "works properly" do
+        expect(User.without_any_colors("white").with_any_colors("blue").order(:created_at).limit(10)).to eq [@user1, @user3]
+      end
     end
   end
 
-  describe "with complex scope" do
-    it "works properly" do
-      expect(User.without_any_colors("white").with_any_colors("blue").order(:created_at).limit(10)).to eq [@user1, @user3]
+  context "with allow_list" do
+    context "when options is not an array" do
+      it "raises an error" do
+        expect { User.acts_as_taggable_array_on :colors, allow_list: "red, blue" }
+          .to raise_error(described_class::InvalidAllowListTypeError)
+      end
+    end
+
+    context "when options is an array" do
+      before do
+        User.acts_as_taggable_array_on :colors, allow_list: %w[red blue]
```

Comment:
Would be interesting to know, what would happen in case of any other datatype passed to allow_list. How can we handle that better?

e.g. string or number:|
`taggable_array :colors, allow_list: 'red' `

or hash:
`taggable_array :colors, allow_list: {red: 'red'}

Reply:
Also would be interesting to know, how would we comparse arrays with strings inside and symbols. I would expect that

`'red'` would equal `:red`

Can we add a test to make sure, that this is a case?

Reply:
> Would be interesting to know, what would happen in case of any other datatype passed to allow_list. How can we handle that better?
> 
> e.g. string or number: `taggable_array :colors, allow_list: 'red' `
> 
> or hash: `taggable_array :colors, allow_list: {red: 'red'}`

About this, there is a check to guarantee that allow list is an array, and if it is not an array, it throws an error saying that allow list must be an array. There is a test for this too https://github.com/tmiyamon/acts-as-taggable-array-on/pull/40/files#diff-ad2deaee3a6d67e0e4d46e72478cb4ea6fc8209a10fa2c244656f3683f2803bcR166

Reply:
> Also would be interesting to know, how would we comparse arrays with strings inside and symbols. I would expect that `['red']` would equal `[:red]`
> 
> Can we add a test to make sure, that this is a case?

I added tests scenarios to this and make sure to deal with those two options as the same